packages/pwa-kit-react-sdk/src/utils/opentelemetry.js

```diff
@@ -0,0 +1,327 @@
+/*
+ * Copyright (c) 2025, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {trace, context, SpanStatusCode} from '@opentelemetry/api'
+import {hrTimeToMilliseconds, hrTimeToTimeStamp} from '@opentelemetry/core'
+import logger from './logger-instance'
+
+const SERVICE_NAME = 'pwa-kit-react-sdk'
+
+function logSpanData(span, event = 'start', res = null) {
+    const spanContext = span.spanContext()
+    const startTime = span.startTime
+    const endTime = event === 'start' ? startTime : span.endTime
+    const duration = event === 'start' ? 0 : hrTimeToMilliseconds(span.duration)
+
+    // Create the span data object that matches the expected format
+    const spanData = {
+        traceId: spanContext.traceId,
+        parentId: span.parentSpanId,
+        name: span.name,
+        id: spanContext.spanId,
+        kind: span.kind,
+        timestamp: hrTimeToTimeStamp(startTime),
+        duration: duration,
+        attributes: {
+            'service.name': SERVICE_NAME,
```

Comment:
If you agree with the previous comment we need to replace all references of `SERVICE_NAME`:
```suggestion
            'service.name': getServiceName(),
```